scipy/stats/_qmc.py

```diff
@@ -951,14 +862,17 @@ def random(self, n=1):
 
         """
         if self.centered:
-            r = 0.5
+            samples = 0.5
         else:
-            r = self.rg_sample((n, self.d))
+            samples = self.rg_sample((n, self.d))
 
-        q = self.rg_integers(low=1, high=n, size=(n, self.d))
+        perms = np.tile(np.arange(1, n + 1), (self.d, 1)).T
+        for i in range(self.d):
+            np.random.shuffle(perms[:, i])
```

Comment:
Sorry, we should use `self.rng` here to use the generator we seed. Also I would add a note about using `axis=0` which will be possible once we drop numpy 1.16. This leads to improvements.

```suggestion
            # axis=0 can be used once numpy 1.16 is dropped
            self.rng.shuffle(perms[:, i])
```

Reply:
Ah, yes we need to use the `rng`, but I do not think we can use `axis=0` here. It shuffles rows of the whole matrix rather than each element of each column. Here, we need to shuffle the elements of each column independently of other columns.

Reply:
Yes 🤦 , they are not independent if we do this. Ok I will change the suggested edit.

Reply:
BTW, shouldn't we transpose the `perms` later to make the memory access pattern better?
```python
perms = np.tile(np.arange(1, n + 1), (self.d, 1))
for i in range(self.d):
    np.random.shuffle(perms[i, :])
perms = perms.T
```
This enables the `np.shuffle` to access sequential memory regions. For example, 
```
[[1,2,3]
 [1,2,3]
 [1,2,3]]
```
is stored in memory (without metadata) as 
```
[1,2,3,1,2,3,1,2,3]
```
thus shuffling along the last axis is more efficient (in case the size of the array is relatively large).

Reply:
Yes good point, I actually tested it and it's a significant improvement. With n=100, d=50 we go from 230 µs to 190 µs.